jao/webservice.py

```diff
@@ -41,35 +41,49 @@ def query_auction_details_by_month(self, corridor: str, month: date, horizon: st
         # prepare the specific input arguments needed, start day the day before the months begin
         # end date the last day of the month
 
-        month_begin = month.replace(day=1)
-        month_end = month.replace(day=monthrange(month.year, month.month)[1])
-        if horizon == 'Yearly':
+        if horizon == "Monthly":
+            month_begin = query_date.replace(day=1)
+            month_end = query_date.replace(day=monthrange(query_date.year, query_date.month)[1])
+            r = self.s.get(self.BASEURL + 'getauctions', params={
+                'corridor': corridor,
+                'fromdate': (month_begin - timedelta(days=1)).strftime("%Y-%m-%d"),
+                'horizon': horizon,
+                'todate': month_end.strftime("%Y-%m-%d"),
+                'shadow': int(shadow_auctions_only)
+            })
+        elif horizon == 'Yearly':
             r = self.s.get(self.BASEURL + 'getauctions', params={
                 'corridor': corridor,
-                'fromdate': month.strftime('%Y-%m-%d'),
+                'fromdate': f"{query_date.year-1}-12-31",
                 'horizon': horizon,
                 'shadow': int(shadow_auctions_only)
             })
         else:
             r = self.s.get(self.BASEURL + 'getauctions', params={
                 'corridor': corridor,
-                'fromdate': (month_begin - timedelta(days=1)).strftime("%Y-%m-%d"),
+                'fromdate': query_date.strftime("%Y-%m-%d-%H:%M:%S"),
                 'horizon': horizon,
-                'todate': month_end.strftime("%Y-%m-%d"),
                 'shadow': int(shadow_auctions_only)
             })
 
+        # improves error feedback a lot
+        if r.status_code >= 400 and not r.reason:
+            r.reason = r.text
         r.raise_for_status()
 
         data = r.json()
-        # pretify the results since we know it is for monthly auction
+        # prettify the results to only show the first products and results
         data = data[0]
         data = {**data, **data['results'][0], **data['products'][0]}
```

Comment:
This returns only the results from the first auction with this ID.
When having daily products, 24 biddings exist with the same ID, which have a different date (one for each hour).
This is currently not reflected in the auctions code.

A split to multiple datasets:

* auctions
* results
* products
would help here, though this is much more work.

I don't know what you think @fboerman - maybe it surely is worth it to provide the option to consistently download data for the different horizons as well here..?

Reply:
hmm that is an interesting one, I dont really have the time to go through this given that I dont use it myself. If you want to take a crack at this feel free!